### PR TITLE
Add min committee size predicate to NewCommittee

### LINKED ISSUE #3666: Validate minCommitteeSize
When proposing a NewCommittee governance action ensure that the number of committee members is at least the minimum number specified in the PParams

### PR DESCRIPTION
**Pull request by aniketd**
# Description

Closes #3666 

This will cause conflicts with #3664 

# Checklist

- [x] Commit sequence broadly makes sense and commits have useful messages
- [x] New tests are added if needed and existing tests are updated
- [x] When applicable, versions are updated in `.cabal` and `CHANGELOG.md` files according to the
      [versioning process](https://github.com/input-output-hk/cardano-ledger/blob/master/RELEASING.md#versioning-process).
- [x] The version bounds in `.cabal` files for all affected packages are updated. **If you change the bounds in a cabal file, that package itself must have a version increase.** (See [RELEASING.md](https://github.com/input-output-hk/cardano-ledger/blob/master/RELEASING.md#versioning-process))
- [x] All visible changes are prepended to the latest section of a `CHANGELOG.md` for the affected packages. **New section is never added with the code changes.** (See [RELEASING.md](https://github.com/input-output-hk/cardano-ledger/blob/master/RELEASING.md#changelogmd))
- [x] Code is formatted with [`fourmolu`](https://github.com/fourmolu/fourmolu) (use `scripts/fourmolize.sh`)
- [x] Cabal files are formatted (use `scripts/cabal-format.sh`)
- [x] [`hie.yaml`](https://github.com/input-output-hk/cardano-ledger/blob/master/hie.yaml) has been updated (use `scripts/gen-hie.sh`)
- [x] Self-reviewed the diff

### CODE REVIEW

File: eras/conway/impl/CHANGELOG.md [hunk 1]
```diff
@@ -2,6 +2,9 @@
 
 ## 1.8.0.0
 
+* Add `MinCommitteeSize` predicate to `NewCommittee` Gov Action #3668
```
Comment (lehins) [inline eras/conway/impl/CHANGELOG.md]: This `MinCommitteeSize` - looks like there is a type with such name, which totally confused me. Something along these lines I think would be better:
```suggestion
* Add `"minCommitteeSize"` pparam validation for `NewCommittee` Gov Action #3668
```

File: eras/conway/impl/CHANGELOG.md [hunk 2]
```diff
@@ -2,6 +2,9 @@
...
+  * Add `committeeMembersL` and `committeeQuorumL` lenses for `Committee`
+  * Add `NewCommitteeSizeTooSmal` `PredicateFailure` in `GOV`
```
Comment (lehins) [inline eras/conway/impl/CHANGELOG.md]: Spelling plus couple more entries:
```suggestion
  * Add `NewCommitteeSizeTooSmall` `PredicateFailure` in `GOV`
  * Add `gePParams` to `GovEnv`
  * Rename `teTxId` -> `geTxId` and `teEpoch` -> `geEpoch`
```

**Review by lehins — APPROVED** [review]
Looks great!

File: eras/conway/impl/CHANGELOG.md (hunk 2 above)
Comment (aniketd) [reply eras/conway/impl/CHANGELOG.md]: Ouch! :hand_over_mouth:

Comment (aniketd) [reply eras/conway/impl/CHANGELOG.md]: The extra changes should now be part of `master` due to #3664 so I didn't add them here again. I will make sure they are part of the changelog if they are missing there :+1:

File: eras/conway/impl/CHANGELOG.md [hunk 3]
```diff
@@ -2,10 +2,14 @@
...
 ## 1.8.0.0
 
+* Add "minCommitteeSize" `PParam` validation for `NewCommittee` `GovAction` #3668
+  * Add `committeeMembersL` and `committeeQuorumL` lenses for `Committee`
+  * Add `NewCommitteeSizeTooSmall` `PredicateFailure` in `GOV`
 * Implement expiry for governance proposals #3664
   * Update `ppGovActionExpiration` to be an `EpochNo`
   * Add `gasExpiresAfter :: !EpochNo` to `GovActionState`
   * Add `gePParams` to `GovEnv`
+  * Rename `teTxId` to `geTxId` and `teEpoch` to `geEpoch`
```
Comment (aniketd) [inline eras/conway/impl/CHANGELOG.md:12]: Note: @lehins I added this here, since the change was part of #3664 I hope this is okay.

Comment (lehins) [reply eras/conway/impl/CHANGELOG.md:12]: Yeah, it doesn't matter too much in which part of the changelog it is.

**Pull request merged**